packages/gatsby-image/src/index.js

```diff
@@ -354,12 +418,20 @@ class Image extends React.Component {
 
           {/* Show the blurry base64 image. */}
           {image.base64 && (
-            <Img src={image.base64} {...placeholderImageProps} />
+            <picture>
+              <source media={image.media} srcSet={image.base64} />
```

Comment:
if we go for the above imageVariants / image variable strategy we could  omit this line as it's part of imageVariants anyway
```suggestion
{generateBase64Sources(imageVariants)}
```

Reply:
? Your suggestion is to remove the source element and add another `generateBase64Sources()` without the `imageVariants` logical guard?

This is the current PR version(since the preview is a bit annoying for context):

```JSX
          {/* Show the blurry base64 image. */}
          {image.base64 && (
            <picture>
              <source media={image.media} srcSet={image.base64} />
              {imageVariants && generateBase64Sources(imageVariants)}
              <Img src={image.base64} {...placeholderImageProps} />
            </picture>
          )}
```

In my earlier review I requested it be separate from variants handling. For more details, see the [related review comment](https://github.com/gatsbyjs/gatsby/pull/13395#discussion_r277518050), also [kind of related](https://github.com/gatsbyjs/gatsby/pull/13395#discussion_r277495092)

Originally there was conditional logic for the 1st element/image in variants, then I made a request to handle it differently. @brimtown mentioned that both sources may be needed and that the order matters.

It's been a while, so I can't recall if there were any other concerns/reasons or the code has evolved since then. It does seem with the current state of code that we can offload all source generation to the methods without any conditional handling?

That aside, if there are no image variants, should an additional source being generated regardless?(I think that was to cater specifically to media prop, if so it could use a logical guard if not moved into generate method)

Reply:
I find
```html
<source media={image.media} srcSet={image.base64} />
{imageVariants && generateBase64Sources(imageVariants)} 
```
a bit weird to read because `<source media={image.media} srcSet={image.base64} />` is also a source tag. We can put the guard inside generateBase64Sources and just return null.